web/ui/ui.go

```diff
@@ -47,6 +47,12 @@ var Assets http.FileSystem = func() http.FileSystem {
 	static := filter.Keep(
 		http.Dir(path.Join(assetsPrefix, "static")),
 		func(path string, fi os.FileInfo) bool {
+			// Disable the experimental React-based UI in OCP builds.
+			// TODO: to be removed when the legacy UI is replaced by the React UI.
```

Comment:
Not opposed to carry this patch, but could this not be contributed to prometheus upstream as well controlled by a flag instead of the current if condition?

Reply:
From an upstream perspective, there's no reason to have a flag (I'm quite sure it would be rejected).
Also the purpose of the patch would be different: instead of not embedding the React assets, it would conditionally expose the `/new*` paths and the React assets would still be present in the Go binary.